apps/src/code-studio/progressRedux.js

```diff
@@ -660,6 +664,19 @@ export const groupedLessons = (state, includeBonusLevels = false) => {
 
   const allLevels = levelsByLesson(state);
 
+  state.lessonGroups.forEach(lessonGroup => {
+    byGroup[lessonGroup.display_name] = {
```

Comment:
in case two lesson groups had the same display name, I think it would make more sense to group by key rather than display name, assuming that information is available here.